cli/azd/internal/repository/infra_confirm.go

```diff
@@ -406,3 +427,49 @@ func (i *Initializer) chooseAuthType(ctx context.Context, serviceName string) (s
 		return scaffold.AuthType_PASSWORD, nil
 	}
 }
+
+func detectCosmosSqlDatabaseContainersInDirectory(root string) ([]scaffold.CosmosSqlDatabaseContainer, error) {
+	var result []scaffold.CosmosSqlDatabaseContainer
+	err := filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.IsDir() && filepath.Ext(path) == ".java" {
+			container, err := detectCosmosSqlDatabaseContainerInFile(path)
+			if err != nil {
+				return err
+			}
+			if len(container.ContainerName) != 0 {
+				result = append(result, container)
+			}
+		}
+		return nil
+	})
+	return result, err
+}
+
+func detectCosmosSqlDatabaseContainerInFile(filePath string) (scaffold.CosmosSqlDatabaseContainer, error) {
```

Comment:
This is something we should re-think about later:
- This approach is not the most efficient way to analyze, if we want to scan cosmos database names and service bus jms queue names at the same time
- Regex can only cover a subset of all the cases

Reply:
Does `later` mean `not in current PR`?

Reply:
Yes, that's what I meant